bootstrap/download-streamine-artifacts.py

```diff
@@ -0,0 +1,178 @@
+		resp_json = json.load(content)
+		entities_maven_deps = filter(lambda x: 'mavenDeps' in x and x['mavenDeps'] != None, resp_json['entities'])
+		maven_deps_list.extend(map(lambda x: x['mavenDeps'], entities_maven_deps))
+	return maven_deps_list
+
+
+def write_downloaded_information_to_output_file(output_file, downloaded_artifacts):
+	with open(output_file, 'w') as fw:
+		for artifact_info, downloaded_location in downloaded_artifacts.iteritems():
+			fw.write("%s\t%s\n" % (artifact_info, downloaded_location))
+
+
+def main():
+	if len(sys.argv) < 3:
+		print("USAGE: [python] %s [streamline config file] [output file]" % sys.argv[0])
+		sys.exit(1)
+
+	config_file = sys.argv[1]
+	output_file = sys.argv[2]
+
+	print("> CONFIG FILE: %s" % config_file)
+	print("> OUTPUT FILE: %s" % output_file)
+
+	config_struct = load_config(config_file)
+
+	print("================ CONFIG ==================")
+	print("CATALOG ROOT URL: %s" % config_struct.catalog_root_url)
+	print("STORM HOME DIR: %s" % config_struct.storm_home_dir)
+	print("MAVEN REPOSITORY URL: %s" % config_struct.maven_repo_url)
+	print("HTTP PROXY URL: %s" % config_struct.http_proxy_url)
+	print("HTTP PROXY USERNAME: %s" % config_struct.http_proxy_username)
+	print("HTTP PROXY PASSWORD: %s" % config_struct.http_proxy_password)
+	print("STORM SUBMIT TOOL CLASSPATH: %s" % config_struct.get_storm_submit_tool_classpath())
+
+	component_bundles_kind = load_component_bundles_kind(config_struct)
+
+	maven_deps_list = collect_maven_deps_list(config_struct, component_bundles_kind)
+
+	print("============== COLLECTED Maven Dependencies ==============")
+	for maven_deps in maven_deps_list:
+		print(maven_deps)
+
+	downloaded_artifacts = {}
+
+	for maven_deps in maven_deps_list:
+		submit_tool_cmd = config_struct.build_storm_submit_tool_command(maven_deps)
+		print("============== Running COMMAND: %s" % submit_tool_cmd)
+		p = subprocess.Popen(submit_tool_cmd, stdout=subprocess.PIPE)
+		stdout, errors = p.communicate()
+		print("============== Return value of SUBMIT TOOL: %s" % p.returncode)
+		if p.returncode != 0:
+			raise RuntimeError("dependency handler returns non-zero code: code<%s> syserr<%s>" % (p.returncode, errors))
+		if not isinstance(stdout, str):
+			stdout = stdout.decode('utf-8')
+		ret_json = json.loads(stdout)
+		downloaded_artifacts.update(ret_json)
+	
+	print("============== Writing downloaded information to file: %s" % output_file)
+	write_downloaded_information_to_output_file(output_file, downloaded_artifacts)
+
+	print("DONE!")
+
+	print("Please follow below steps:")
+	print("1. archive directory of either '.m2/repository' in home directory, or 'local-repo' in current directory based on location for downloaded artifacts.")
+	print("2. send archived file to the machines which streamline is installed and don't have access to internet.")
+	print("3. extract file to '.m2/repository' in home directory of streamline account. Please make the directory if it doesn't exist.")
```

Comment:
Sometimes the artifacts are in local_repo directory. Does it always fall back to .m2/repository? How about the jar permission and ownership - will it have required permissions for the user streamline is running as ?

Reply:
| .m2/repository | local-repo | download location |
| X | X | local-repo |
| O | X | .m2/repository |
| X | O | local-repo |
| O | O | .m2/repository |

So whenever '.m2/repository' directory exists, the directory is used. 'local-repo' is used for failing back.

For permission and ownership, I guess it belongs to the account which runs the script. So simplest approach is run this script as the account which runs streamline. Maybe just ensuring  '.m2/repository' exists should make it simpler. 

One thing we need to consider is, downloaded artifacts will be archived and copied to the other machine and extracted, so final ownership of archives will belong to the account which extracts the archive.

Could you suggest how to document this? I'm not sure how much the script should guide to users kindly.